File: 8-kyu/sum-of-positive.js
// https://www.codewars.com/kata/sum-of-positive/train/javascript

function positiveSum(arr) {
  if (arr.length < 1) return 0
  let result = 0
  for (let i = 0; i < arr.length; i++) {
    const num = arr[i];
    if (num > 0) {
      result += num
    }
  }
  return result
}

function positiveSum(arr) {
  return arr.reduce((a, b) => a + (b > 0 ? b : 0), 0)
}

console.log(positiveSum([1,2,3,4,5]),15);
console.log(positiveSum([1,-2,3,4,5]),13);
console.log(positiveSum([]),0);
console.log(positiveSum([-1,-2,-3,-4,-5]),0);
console.log(positiveSum([-1,2,3,4,-5]),9);

File: rust/8-kyu/october/sum-of-positive.rs
// https://www.codewars.com/kata/5715eaedb436cf5606000381/train/rust

fn positive_sum(slice: &[i32]) -> i32 {
    let mut sum = 0;

    for &n in slice {
        if n > 0 {
            sum += n;
        }
    }

    sum
}

fn positive_sum(slice: &[i32]) -> i32 {
    slice
        .iter()
        .filter(|x| x.is_positive())
        .sum()
}

fn positive_sum(slice: &[i32]) -> i32 {
    slice
        .iter()
        .filter(|&&x| x > 0)
        .sum()
}
